includes/data-events/class-data-events.php

```diff
@@ -0,0 +1,230 @@
+	 */
+	public static function register_handler( $handler, $action_name = null ) {
+		/** If first argument is callable, treat as global handler. */
+		if ( empty( $action_name ) ) {
+			self::$global_handlers[] = $handler;
+			return;
+		}
+		if ( ! isset( self::$actions[ $action_name ] ) ) {
+			return new WP_Error( 'action_not_registered', __( 'Action not registered.', 'newspack' ) );
+		}
+		if ( ! is_callable( $handler ) ) {
+			return new WP_Error( 'handler_not_callable', __( 'Handler is not callable.', 'newspack' ) );
+		}
+		if ( in_array( $handler, self::$actions[ $action_name ], true ) ) {
+			return new WP_Error( 'handler_already_registered', __( 'Handler already registered.', 'newspack' ) );
+		}
+		self::$actions[ $action_name ][] = $handler;
+	}
+
+	/**
+	 * Get a list of all registered actions.
+	 *
+	 * @return string[] Registered actions.
+	 */
+	public static function get_actions() {
+		return array_keys( self::$actions );
+	}
+
+	/**
+	 * Get a list of all registered action handlers.
+	 *
+	 * @param string $action_name Action name.
+	 *
+	 * @return callable[] Registered action handlers.
+	 */
+	public static function get_action_handlers( $action_name ) {
+		if ( ! isset( self::$actions[ $action_name ] ) ) {
+			return [];
+		}
+		return self::$actions[ $action_name ];
+	}
+
+	/**
+	 * Whether an action is registered.
+	 *
+	 * @param string $action_name The action name.
+	 *
+	 * @return bool
+	 */
+	public static function is_action_registered( $action_name ) {
+		return isset( self::$actions[ $action_name ] );
+	}
+
+	/**
+	 * Dispatch an action event.
+	 *
+	 * @param string  $action_name   Action name.
+	 * @param array   $data          Data to pass to the action.
+	 * @param boolean $use_client_id Whether to use the session's client ID. Default is true.
+	 *
+	 * @return void|WP_Error Error if action not registered.
+	 */
+	public static function dispatch( $action_name, $data, $use_client_id = true ) {
+		if ( ! self::is_action_registered( $action_name ) ) {
+			return new WP_Error( 'newspack_data_events_action_not_registered', __( 'Action not registered.', 'newspack' ) );
+		}
+
+		$timestamp = time();
+		$client_id = null;
+		if ( $use_client_id ) {
+			$client_id = Reader_Activation::get_client_id();
+		}
+
+		do_action( 'newspack_data_event_dispatch', $action_name, $timestamp, $data, $client_id );
+
+		do_action( "newspack_data_event_dispatch_{$action_name}", $timestamp, $data, $client_id );
+
+		$url = \add_query_arg(
+			[
+				'action' => self::ACTION,
+				'nonce'  => \wp_create_nonce( self::ACTION ),
+			],
+			\admin_url( 'admin-ajax.php' )
+		);
+
+		$body = [
+			'action_name' => $action_name,
+			'timestamp'   => $timestamp,
+			'data'        => $data,
+			'client_id'   => $client_id,
+		];
+
+		return \wp_remote_post(
+			$url,
+			[
+				'timeout'   => 0.01,
+				'blocking'  => false,
+				'body'      => $body,
+				'cookies'   => $_COOKIE, // phpcs:ignore
+				'sslverify' => false,
```

Comment:
WP core offers a filter for people to change it and we could do the same... this could probably be

`'sslverify' => apply_filters( 'https_local_ssl_verify', false ),`

Reply:
Done in 831f51be3841cad0be6ecaf1a901d48e5789db5f